src/Compilers/Shared/CoreClrAnalyzerAssemblyLoader.cs

```diff
@@ -8,19 +8,29 @@ namespace Microsoft.CodeAnalysis
 {
     internal sealed class CoreClrAnalyzerAssemblyLoader : AnalyzerAssemblyLoader
     {
-        private readonly AssemblyLoadContext _loadContext;
+        private AssemblyLoadContext _loadContext;
 
         public CoreClrAnalyzerAssemblyLoader()
         {
-            _loadContext = AssemblyLoadContext.GetLoadContext(typeof(CoreClrAnalyzerAssemblyLoader).GetTypeInfo().Assembly);
+        }
 
-            _loadContext.Resolving += (context, name) =>
+        protected override Assembly LoadFromPathImpl(string fullPath)
+        {
+            // Initialize the _loadContext in lazy fashion
+            if (_loadContext == null)
```

Comment:
@tmat, for that to happen the caller need to be aware of the fact that ALC is not supported either through a flag or trying to GetLoadContext. I will have it like this for now.